src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs

```diff
@@ -0,0 +1,339 @@
+                }
+                terminal.WriteLine();
+            }
+        }
+
+        public void UpdateRegistrySettings(string domainName, string userName)
+        {
+            ShowAutoLogonWarningIfAlreadyEnabled(domainName, userName);
+
+            //machine specific
+            UpdateMachineSpecificRegistrySettings(domainName, userName);
+
+            //user specific
+            UpdateUserSpecificRegistrySettings(domainName, userName);
+        }
+
+        public void RevertRegistrySettings(string domainName, string userName)
+        {
+            //machine specific
+            var hive = RegistryHive.LocalMachine;
+            
+            RevertOriginalValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonUserName);
+            RevertOriginalValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonDomainName);
+            RevertOriginalValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonPassword);
+            RevertOriginalValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonCount);
+            RevertOriginalValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogon);
+            RevertOriginalValue(hive, RegistryConstants.SubKeys.ShutdownReasonDomainPolicy, RegistryConstants.ValueNames.ShutdownReason);
+            RevertOriginalValue(hive, RegistryConstants.SubKeys.ShutdownReasonDomainPolicy, RegistryConstants.ValueNames.ShutdownReasonUI);
+            RevertOriginalValue(hive, RegistryConstants.SubKeys.LegalNotice, RegistryConstants.ValueNames.LegalNoticeCaption);
+            RevertOriginalValue(hive, RegistryConstants.SubKeys.LegalNotice, RegistryConstants.ValueNames.LegalNoticeText);
+
+            //user specific            
+            if (!_windowsServiceHelper.HasActiveSession(domainName, userName))
+            {
+                string securityId = _windowsServiceHelper.GetSecurityId(domainName, userName);
+                RevertRegistrySettingsForDifferentUser(securityId);
+            }
+            else
+            {
+                RevertRegistrySettingsForCurrentUser();
+            }
+        }
+
+        public bool DoesRegistryExistForUser(string domainName, string userName)
+        {
+            if (!_windowsServiceHelper.HasActiveSession(domainName, userName))
+            {
+                string securityId = _windowsServiceHelper.GetSecurityId(domainName, userName);
+                return _registryManager.RegsitryExists(securityId);
+            }
+            return true;
+        }
+
+        private List<string> GetWarningsForUserSpecificSettings(string domainName, string userName)
+        {
+            string screenSaverValue = null;
+            List<string> warningReasons = new List<string>();
+
+            if (!_windowsServiceHelper.HasActiveSession(domainName, userName))
```

Comment:
There is duplicate code all over this file. Why not just use the sid everywhere?